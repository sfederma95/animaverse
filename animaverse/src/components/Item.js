import React from 'react';

function Item({src,action,name,id,description,amount}){
    return(
        <div>
            <img alt={name} src={src}/>
            <p>Item Name: {name}</p>
            <p>Description: {description}</p>
            <p>Effect: {action==="Feed"? "Lower hunger" : "Increase happiness"} by {amount} points</p>
        </div>
    )
}

export default Item;