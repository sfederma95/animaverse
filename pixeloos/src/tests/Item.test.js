import React from 'react';
import {render} from '@testing-library/react';
import {MemoryRouter} from 'react-router';
import Item from '../components/Item';
import {UserProvider} from './testSetup';

it ('renders item component without crashing', function(){
    render(
        <MemoryRouter>
            <UserProvider>
                <Item action='Feed' name='candy' id={1} description='sweet' amount={10} />
            </UserProvider>
        </MemoryRouter>
    )
})

it ('matches snapshot', function(){
    const {asFragment} = render (
        <MemoryRouter>
            <UserProvider>
                <Item action='Feed' name='candy' id={1} description='sweet' amount={10}/>
            </UserProvider>
        </MemoryRouter>
    )
    expect(asFragment()).toMatchSnapshot();
})