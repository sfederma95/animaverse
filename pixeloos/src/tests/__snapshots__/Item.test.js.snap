// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="inv-item"
    id="1"
  >
    <img
      alt="candy"
      class="item-img"
      tabindex="0"
    />
  </div>
</DocumentFragment>
`;
